mitodl/mit-open
# only run black via pre-commit

# What are the relevant tickets?
None

# Description (What does it do?)
This PR removes `black` from our python dependencies and github actions ci. Instead, we will only run black via `pre-commit` and will run it on CI via the pre-commit CI action.

This solves an issue you can see in #171:
- `black` got updated in poetry.lock
- but NOT in pre-commit.yaml

The two versions have slightly different preferences, so we have two CI checks, exactly one of which will pass.

# How can this be tested?
I guess check that pre-commit CI is running black. You can see it is here: https://results.pre-commit.ci/run/github/672068771/1699568416.AGsRY5f1QhOyeYOB-Cscog
